### PR TITLE
PIL-943 Fixed when changing UPE to non uk company in CYA

### PR DESCRIPTION
**Pull request by mrah93**

### CODE REVIEW

**Review by AKKUM — DISMISSED** [review]

File: app/navigation/UltimateParentNavigator.scala [hunk 1]
```diff
@@ -70,8 +75,13 @@ class UltimateParentNavigator @Inject() {
       .getOrElse(routes.JourneyRecoveryController.onPageLoad())
 
   private val checkRouteMap: Page => UserAnswers => Call = {
-    case UpePhonePreferencePage => telephoneCheckRouteLogic
-    case _                      => whichCheckYourAnswerPageContact
+    case UpeRegisteredInUKPage    => domesticOrNotRouteInCheckMode
```
Comment (jassalrichy) [inline app/navigation/UltimateParentNavigator.scala:78]: We could call domesticOrNotRoute directly:
case UpeRegisteredInUKPage    => ua => domesticOrNotRoute(ua, CheckMode)

File: app/navigation/UltimateParentNavigator.scala [hunk 2]
```diff
@@ -35,7 +35,7 @@ class UltimateParentNavigator @Inject() {
   private lazy val reviewAndSubmitCheckYourAnswers = controllers.routes.CheckYourAnswersController.onPageLoad
   private lazy val upeCheckYourAnswers             = controllers.registration.routes.UpeCheckYourAnswersController.onPageLoad
   private val normalRoutes: Page => UserAnswers => Call = {
-    case UpeRegisteredInUKPage    => domesticOrNotRoute
+    case UpeRegisteredInUKPage    => domesticOrNotRouteInNormalMode
```
Comment (jassalrichy) [inline app/navigation/UltimateParentNavigator.scala:38]: We could call domesticOrNotRoute directly:
case UpeRegisteredInUKPage    => ua => domesticOrNotRoute(ua, NormalMode)

File: app/navigation/UltimateParentNavigator.scala (hunk 1 above)
Comment (mrah93) [reply app/navigation/UltimateParentNavigator.scala]: Done

File: app/navigation/UltimateParentNavigator.scala (hunk 2 above)
Comment (mrah93) [reply app/navigation/UltimateParentNavigator.scala]: Done

**Review by jassalrichy — APPROVED** [review]

**Review by javed-iqbal1 — DISMISSED** [review]

**Review by TevynAllen — APPROVED** [review]

**Review by MarjanMubarok — APPROVED** [review]

**Pull request merged**